karinhogenbirk/famousProgrammers
# Cleanup and refactor example

# What this pullrequest does: 

- Cleans up the code in various ways (as an example)
- Prepares the app for deployment on render.com

![quiz-demo](https://user-images.githubusercontent.com/20372832/198015136-d94497e7-fce4-48b4-9e01-f69dbf6fdb3f.gif)

## Why 

The first step is to make things work, but then we want to clean up the project so we can continue to work on it and not get lost all the time. We want to deploy v1 to render before adding a DB -> this simulates improving an app as you go. 

## How

Here are some things I cleaned up, view the video for more details:

- Split up: scraper, seeding, question utils
- Rewrite functions to be reusable
- Split up answer enpoints: `POST /questions/programmers/answer` & `POST /questions/projects/answer`
- Simplify answer logic: find the right programmer once, then check the answer, count the votes and respond

Added changes for deployment to render.com

- Localhost in the url is not allowed!
- Solution: Express hosts html files and images (express.static)

## How to test this

- Run the server on your computer, play the quiz. It should work as before
- You can now change the `PORT` variable to _any_ port: 4000, 9001, etc. And the app should still work without having to change any urls

## Anything else

- Cleaning up can be challenging to do in the beginning, that's why I made the video for you, but you should be **super** proud of your initial solution! You made it work!
